File: hw/arias/U3/Hw_37_Strategy_Pattern/SortApp/src/main/java/ec/edu/espe/sortapp/model/SortingContext.java

package ec.edu.espe.sortapp.model;

/**
 *
 * @author Alan Arias
 */
public class SortingContext {
    private SortingStrategy ss;

    public int[] sort(int data[]) {
        int size = data.length;
        ss = setSortStrategy(size);
        return ss.sort(data);
    }

    private SortingStrategy setSortStrategy(int n) {
        if (n > 0 && n <= 30)
            return new BubbleSort();
        if (n > 30 && n < 100)
            return new InsertionSort();
        return new QuickSort();
    }
}

File: EXAM/sandoval/u3/SandovalStrategy/sorting_context.py
from sorting_strategies import SortingStrategy

class SortingContext:
    def __init__(self, strategy: SortingStrategy):
        self.strategy = strategy
    
    def set_strategy(self, strategy: SortingStrategy):
        self.strategy = strategy
    
    def sort(self, data):
        return self.strategy.sort(data)